iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -168,7 +168,7 @@ public override async Task WriteAsync(IChannelHandlerContext context, object dat
                     throw new IotHubCommunicationException("MQTT is disconnected.");
                 }
 
-                if (data is Message message)
+                if (data is MessageBase message)
```

Comment:
the transport layer should work with the base message class, which has all of the required functionality.